flake.nix

```diff
@@ -193,14 +193,14 @@
 
           lib = {
             mkDevShell =
-              args: pkgs.mkShell {
+              { providers, packages ? [] }: pkgs.mkShell {
                 buildInputs = lib.attrValues
                   (pkgs.callPackage ./nix/devshell.nix
                     {
                       generateSchema = self.generateSchema.${system};
                       nickel = inputs.nickel.packages.${system}.nickel-lang-cli;
                     }
-                    args) ++ [
+                    { inherit providers; }) ++ packages ++ [
```

Comment:
If I'm understanding correctly, this makes it possible to only forward `{providers = ...;}` to `./nix/devshell.nix`. However this package seems to accept an additional argument,  `extraNickelInput`. I don't know if  that is supposed to be useful in a devshell, but for the sake of backward compatibility, should we forward this argument to `./nix/devshell.nix` as well? Another possibility is to just extract packages out and pass everything else to the underlying shell, but I would intuitively err on the side of the former solution, although a bit more repetitive.

Reply:
fair, thanks for the feedback! i pushed an update to reflect this now.